directives/animate/animate.js

```diff
@@ -0,0 +1,250 @@
+import { directive, PropertyPart } from 'lit-html';
+import { getComposedActiveElement, getNextFocusable } from '../../helpers/focus.js';
+import { getComposedParent, isComposedAncestor } from '../../helpers/dom.js';
+
+const stateMap = new WeakMap();
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+const showTransitionDuration = 300;
+const hideTransitionDuration = 200;
+const moveYValue = 20;
+
+class AnimationState {
+
+	constructor(propertyPart) {
+
+		if (!(propertyPart instanceof PropertyPart) || propertyPart.committer.name !== 'animate') {
+			throw new Error('animation directives must be used with "animation" property');
+		}
+
+		this.id = 0;
+		this.clone = null;
+		this.elem = propertyPart.committer.element;
+		this.state = 'unknown';
+
+	}
+
+	async animate(info) {
+
+		if (this.clone === null) {
+			this.clone = document.createElement('div');
+		}
+		this.elem.parentNode.insertBefore(this.clone, this.elem);
```

Comment:
I think consumers ought to be conscious of this, since this might cause their `slotchange` event to be dispatched.  I suspect there is some code that would not be resilient to this... so something we need to be mindful of when building components that assume certain slot content not contain extraneous elements.

Reply:
Agreed yeah. I don't see a way around this, since the clone is the thing that has to be present in the DOM taking up space as the host is transitioned in/out. Might just be a case of "fix places that break as they come up".

Reply:
Cool, yeah.  I think this is something we should just be mindful of when writing/reviewing code that wires up to `slotchange`.  Not that everything needs to handle this case, but just thinking about how the such handlers will deal with these elements if there is a chance the consumer would use these directives in those slots.